plugins/upstash/api_key.go

```diff
@@ -0,0 +1,81 @@
+package upstash
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://docs.upstash.com/redis/account/developerapi#create-an-api-key"), // TODO: Replace with actual URL
+		ManagementURL: sdk.URL("https://console.upstash.com/account/api"), 
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Upstash.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 36,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Email,
+				MarkdownDescription: "Email used to authenticate to Upstash.",
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+		},
+        DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryUpstashConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"UPSTASH_API_KEY": fieldname.APIKey, 
+	"UPSTASH_EMAIL" : fieldname.Email,
+}
+
+func TryUpstashConfigFile() sdk.Importer {
+	return importer.TryFile("~/.upstash.json", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		var config Config
+		if err := contents.ToYAML(&config); err != nil {
+			out.AddError(err)
+			return
+		}
```

Comment:
I see you are opening a json file, but parsing a yaml one. Which one is it?

Reply:
json file

Reply:
I'll mark this conversation as resolved because we are now properly sing `.ToJSON` and [functionally, the shell plugin works](https://github.com/1Password/shell-plugins/pull/316#pullrequestreview-1505690318).